src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -180,20 +259,86 @@ def __init__(
         # store the context in the process of assemble the AgentPool object
         self.context = AKSAgentPoolContext(cmd, raw_parameters, self.models, decorator_mode=DecoratorMode.CREATE)
 
+    def _ensure_agentpool(self, agentpool: AgentPool) -> None:
+        """Internal function to ensure that the incoming `agentpool` object is valid and the same as the attached
+        `agentpool` object in the context.
+
+        If the incomding `agentpool` is not valid or is inconsistent with the `agentpool` in the context, raise a
+        CLIInternalError.
+
+        :return: None
+        """
+        if not isinstance(agentpool, self.models.AgentPool):
+            raise CLIInternalError(
+                "Unexpected agentpool object with type '{}'.".format(type(agentpool))
+            )
+
+        if self.context.agentpool != agentpool:
```

Comment:
this will check the memory equal or the value equal?

Reply:
Actually this is value equal. The base class of the model provided by the SDK overrides the way to compare equality, that is, to compare values.